src/modules/escrow/services/escrow.transaction.service.ts

```diff
@@ -1,41 +1,48 @@
-import { Address, AddressValue, TokenTransfer } from '@multiversx/sdk-core/out';
+import {
+    Address,
+    AddressValue,
+    Token,
+    TokenTransfer,
+} from '@multiversx/sdk-core';
 import { Injectable } from '@nestjs/common';
-import BigNumber from 'bignumber.js';
-import { gasConfig, mxConfig } from 'src/config';
+import { gasConfig } from 'src/config';
 import { InputTokenModel } from 'src/models/inputToken.model';
 import { TransactionModel } from 'src/models/transaction.model';
+import { TransactionOptions } from 'src/modules/common/transaction.options';
 import { MXProxyService } from 'src/services/multiversx-communication/mx.proxy.service';
 
 @Injectable()
 export class EscrowTransactionService {
     constructor(private readonly mxProxy: MXProxyService) {}
 
     async withdraw(senderAddress: string): Promise<TransactionModel> {
-        const contract = await this.mxProxy.getEscrowContract();
-
-        return contract.methodsExplicit
-            .withdraw([new AddressValue(Address.fromString(senderAddress))])
-            .withChainID(mxConfig.chainID)
-            .withGasLimit(gasConfig.escrow.withdraw)
-            .buildTransaction()
-            .toPlainObject();
+        return await this.mxProxy.getEscrowSmartContractTransaction(
+            new TransactionOptions({
+                sender: senderAddress,
+                gasLimit: gasConfig.escrow.withdraw,
+                function: 'withdraw',
+                arguments: [
+                    new AddressValue(Address.newFromBech32(senderAddress)),
+                ],
+            }),
+        );
     }
 
     async cancelTransfer(
         senderAddress: string,
         receiverAddress: string,
     ): Promise<TransactionModel> {
-        const contract = await this.mxProxy.getEscrowContract();
-
-        return contract.methodsExplicit
-            .cancelTransfer([
-                new AddressValue(Address.fromString(senderAddress)),
-                new AddressValue(Address.fromString(receiverAddress)),
-            ])
-            .withChainID(mxConfig.chainID)
-            .withGasLimit(gasConfig.escrow.cancelTransfer)
-            .buildTransaction()
-            .toPlainObject();
+        return await this.mxProxy.getEscrowSmartContractTransaction(
+            new TransactionOptions({
+                sender: senderAddress,
```

Comment:
`senderAddress` is not the actual sender for this transaction. This transaction is done by an admin user and it requires 2 arguments for `cancelTransfer` endpoint: `senderAddress` and `receiverAddress`. You still need to add the `userAddress` as argument for this method which will be the admin user

Reply:
done